### PR TITLE
Replace GTimeVal by struct timespec

### PR DESCRIPTION
**Pull request by ylse**
GTimeVal in glib-2.0 is being deprecated since 2.62 because it is not
year-2038-safe. Replace it by struct timespec.

Signed-off-by: Ylse <27207371+ylse@users.noreply.github.com>

### CODE REVIEW

**Comment by kira-syslogng** [comment]
This user does not have permission to start the build. Can one of the admins verify this patch and start the build?
(admin: you have the next options (make sure you checked the code):
"ok to test" to accept this pull request (and further changes) for testing
"test this please" for a one time test run
"add to whitelist" add author of a Pull Request to whitelist (globally, be careful, it means this user can trigger kira for any PR)
do nothing -> CI won't start)

**Comment by kira-syslogng** [comment]
This user does not have permission to start the build. Can one of the admins verify this patch and start the build?
(admin: you have the next options (make sure you checked the code):
"ok to test" to accept this pull request (and further changes) for testing
"test this please" for a one time test run
"add to whitelist" add author of a Pull Request to whitelist (globally, be careful, it means this user can trigger kira for any PR)
do nothing -> CI won't start)

**Comment by alltilla** [comment]
@kira-syslogng test this please

**Comment by bazsi** [comment]
Can you order commits so that each would compile? that might mean that a lot of these needs to be squashed into a single commit (e.g. cached_g_current_time() prototype change will cause most of these changes to go into a single commit)

This would make it easier to bisect later, also it makes it more difficult to review, as we have non-compiling patches.

As a followup to this idea, it'd be great if we could increase UnixTimeStamp granularity to nanoseconds, at the moment it's microseconds only.

File: lib/logqueue.c [hunk 1]
```diff
@@ -185,13 +185,13 @@ log_queue_check_items(LogQueue *self, gint *timeout, LogQueuePushNotifyFunc para
     {
       gint64 diff;
       gint new_buckets;
-      GTimeVal now;
+      struct timespec now;
 
-      g_get_current_time(&now);
+      timespec_get(&now, TIME_UTC);
```
Comment (bazsi) [inline lib/logqueue.c:190]: I would use clock_gettime() if possible. Honestly I didn't even know about timespec_get() and was looking for its implementation when I had to realize this is standard too. But seems to be more recent than clock_gettime() being part of C11, whereas clock_gettime() is from the 90s.

Also, I find it strange that I need to specify a timezone to an epoch based timestamp.

**Comment by bazsi** [comment]
I am working on a somewhat different implementation of the same in #4741, so closing this instead of that.

**Pull request closed without merging**